File: user-service/src/main/java/com/gallery/userservice/infrastructure/AuthEntity.java
package com.gallery.userservice.infrastructure;

import com.gallery.userservice.domain.Auth;
import jakarta.persistence.*;

@Entity
@Table(name = "AUTH")
public class AuthEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "UserId", nullable = false)
    private int userId;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(name = "PasswordHash", nullable = false)
    private String passwordHash;

    public AuthEntity() {}

    public AuthEntity(Auth auth) {
        this.id = auth.getId();
        this.userId = auth.getUserId();
        this.email = auth.getEmail();
        this.passwordHash = auth.getPasswordHash();
    }

    public Auth toDomain() {
        return new Auth(
                this.id,
                this.userId,
                this.email,
                this.passwordHash
        );
    }

    // Getters & Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPasswordHash() { return passwordHash; }
    public void setPasswordHash(String passwordHash) { this.passwordHash = passwordHash; }
}

File: app/UserService/Infrastructure/Entities/AuthEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using UserService.Domain;

namespace UserService.Infrastructure.Entities
{
    [Table("AUTH", Schema = "dbo")]
    public class AuthEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("Id")]
        public int Id { get; set; }

        [Column("UserId")]
        [Required]
        public int UserId { get; set; }

        [Column("Email")]
        [Required]
        [StringLength(100)]
        public string Email { get; set; }

        [Column("PasswordHash")]
        [Required]
        [StringLength(255)]
        public string PasswordHash { get; set; }

        public AuthEntity()
        {
            Id = 0;
            UserId = 0;
            Email = "";
            PasswordHash = "";
        }

        public AuthEntity(Auth auth)
        {
            Id = auth.Id;
            UserId = auth.UserId;
            Email = auth.Email;
            PasswordHash = auth.PasswordHash;
        }

        public Auth ToAuth()
        {
            return new Auth(Id, UserId, Email, PasswordHash);
        }
    }
}
